pulser/simulation/simulation.py

```diff
@@ -15,116 +15,214 @@
+            raise ValueError("No instructions given for the channels in the "
+                             "sequence.")
+        self._seq = seq
+
+    @property
+    def config(self) -> SimConfig:
+        """Property getter for config."""
+        return self._config
+
+    @config.setter
+    def config(self, cfg: SimConfig) -> None:
+        self._config = cfg
+        self._set_param_from_config()
+
+    def show_config(self) -> None:
+        """Shows current configuration."""
+        print(self._config)
+
+    def reset_config(self) -> None:
+        """Resets configuration to default."""
+        self.config = SimConfig()
+        print("Configuration has been set to default.")
 
-        if isinstance(evaluation_times, str):
-            if evaluation_times == 'Full':
-                self.eval_times = deepcopy(self._times)
-            elif evaluation_times == 'Minimal':
-                self.eval_times = np.array([self._times[0], self._times[-1]])
+    def _set_param_from_config(self) -> None:
+        """Sets all relevant Simulation parameters from its SimConfig.
+
+        Called every time a new configuration is loaded in order to update
+        Simulation parameters.
+        """
+        self._prepare_bad_atoms()
+        # update samples with potential noise settings
+        self._build_hamiltonian_from_seq()
+        if 'dephasing' in self.config.noise:
+            self._init_dephasing()
+
+    def _prepare_bad_atoms(self) -> None:
+        """Chooses atoms that will be badly prepared.
+
+        If no SPAM noise, simply sets all atoms as well-prepared.
+        """
+        # Tag True if atom is badly prepared
+        self._bad_atoms: dict[Union[str, int], bool] = {
+            qid: False for qid in self._qid_index}
+        if 'SPAM' in self.config.noise:
+            self._bad_atoms = \
+                {qid: (np.random.uniform() < self.config.spam_dict['eta'])
+                 for qid in self._qid_index}
+
+    @property
+    def initial_state(self) -> qutip.Qobj:
+        """Property getter for initial_state."""
+        return self._initial_state
+
+    @initial_state.setter
+    def initial_state(self, value: Union[str, ArrayLike, qutip.qObj]) -> None:
+        self._initial_state: qutip.Qobj
+        if value == 'all-ground':
+            self._initial_state = \
+                qutip.tensor([self.basis['g'] for _ in range(self._size)])
+        else:
+            if isinstance(value, qutip.Qobj):
+                if cast(qutip.Qobj, value).shape != \
+                        (self.dim**self._size, 1):
+                    raise ValueError("Incompatible shape of initial_state")
+                self._initial_state = value
+            else:
+                if cast(np.ndarray, value).shape != \
+                        (self.dim**self._size,):
+                    raise ValueError("Incompatible shape of initial_state")
+                self._initial_state = qutip.Qobj(value)
```

Comment:
These two are very similar, this can probably be refactored to avoid duplication

Reply:
(I meant the two checks that raise the same ValueError)